tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -286,6 +298,45 @@ def _fit(self,
 
             self.params_trained['best_epoch'] = save_callback.best_epoch
 
+    def _get_batch_size(self, X, default_batch_size_for_small_inputs=32):
+        bs = self.params['bs']
+        if bs == 'auto':
+            bs = 512 if len(X) >= 200000 else 256
+        bs = bs if len(X) > bs else default_batch_size_for_small_inputs
+
+        if self.params['bs'] == 'auto':
+            logger.log(15, f'Automated batch size selection: {bs}')
+
+        return bs
+
+    def _get_epochs_number(self, samples_num, epochs, batch_size, time_left=None, min_batches_count=30, default_epochs=30):
+        if epochs == 'auto':
+            batches_count = int(samples_num / batch_size) + 1
+            if not time_left:
+                return default_epochs
+            elif batches_count < min_batches_count:
+                return default_epochs
+            else:
+                est_batch_time = self._measure_batch_times(min_batches_count)
+                est_epoch_time = batches_count * est_batch_time * 1.1
+                est_max_epochs = int(time_left / est_epoch_time)
+                epochs = min(default_epochs, est_max_epochs)
+                logger.log(15, f'Automated epochs selection: training for {epochs} epoch(s). Estimated time budget use {epochs * est_epoch_time:.2f} / {time_left:.2f} sec')
+        return epochs
+
+    def _measure_batch_times(self, min_batches_count):
+        from fastai.callback.core import CancelFitException
+        # from .callbacks import BatchTimeTracker
+        batch_time_tracker_callback = BatchTimeTracker(batches_to_measure=min_batches_count)
```

Comment:
Why is this commented out? BatchTimeTracker is never imported yet it is used in the next line.

Reply:
```
NameError: name 'BatchTimeTracker' is not defined
```

Reply:
fixed